Frontend/nutri-pal/src/app/mealregistration/search/page.tsx

```diff
@@ -0,0 +1,90 @@
+'use client'
+
+import { BASE_URL_EDAMAME_SEARCH_API, BASE_URL_REST_API } from "@/app/consts";
+import { MealRegistrationForm } from "@/components/mealRegister/MealRegistrationForm"
+import { SearchResults } from "@/components/mealRegister/SearchResults";
+import { TopBarMain } from "@/components/navigation/TopBarMain"
+import '@/styles/searchMeal.css'
+import axios from "axios";
+import { useEffect, useState } from "react";
+import { IoSearchSharp } from "react-icons/io5";
+
+
+export default function search() {
+    const SEARCH_ICON_SIZE = 20;
+
+    const [searchQuery, setSearchQuery] = useState('');
+    const [foods, setFoods] = useState<any>([]);
+    const [page, setPage] = useState(0);
+
+    // Searches food by query from Edamame API, returns array of 20 foods with nutrients
+    const searchFoods = () => {
+        axios.get(BASE_URL_EDAMAME_SEARCH_API + `&ingr=${searchQuery}` + `&page=${page}`)
+            .then((response: { data: any }) => {
+                if (foods === null) {
+                    setFoods(response.data.hints)
+                } else {
+                    setFoods([...foods, ...response.data.hints])
```

Comment:
Checking your understanding: why do you need to append data to a list like this? 

https://immutable-js.com/

& to get around this, you could use https://immutable-js.com/ . This way you don't need to remember to use this instead of list.push